### PR TITLE
Exit gracefully on SIGTERM

### PR DESCRIPTION
**Pull request by gbenhaim**
This PR should fix https://github.com/lago-project/lago/issues/669 by catching SIGTERM and thus require cleanup actions.

### CODE REVIEW

File: lago/cmd.py [hunk 1]
```diff
@@ -899,7 +900,18 @@ def create_parser(cli_plugins, out_plugins):
...
+def signal_handler(signum, frame):
+
+    LOGGER.debug('Signal {} was caught'.format(signum))
+
+    if signum == signal.SIGTERM:
+        sys.exit(128 + signum)
+
+
 def main():
+
```
Comment (ifireball) [inline lago/cmd.py]: Maybe add a comment here about why is this needed

File: lago/cmd.py [hunk 2]
```diff
@@ -899,7 +900,18 @@ def create_parser(cli_plugins, out_plugins):
     return parser
 
 
+def signal_handler(signum, frame):
+
```
Comment (ifireball) [inline lago/cmd.py]: "signal_handler" is a too generic name, won't it be better to call it "sigterm_handler", or better, just give it a name that tells what it does, so the link between the signal and the activity it triggers becomes obvious in the 'signal.signal' line.

File: lago/cmd.py [hunk 3]
```diff
@@ -27,6 +27,7 @@
 import sys
 from textwrap import dedent
 import warnings
+import signal
```
Comment (ifireball) [inline lago/cmd.py]: Maybe its better to be explicit here, and do a "from signal import ..."

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by gbenhaim** [comment]
@ifireball done.

File: lago/cmd.py [hunk 4]
```diff
@@ -899,7 +900,28 @@ def create_parser(cli_plugins, out_plugins):
...
+    """
+
+    LOGGER.debug('SIGTERM was caught')
+    sys.exit(128 + signum)
+
+
 def main():
+
+    # Trigger cleanup on SIGTERM
+    signal(SIGTERM, sigterm_handler)
```
Comment (danielBelenky) [inline lago/cmd.py]: I think you should catch SIGHUP as well just for cases where Lago is used on a remote host and connection closes.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by gbenhaim** [comment]
@danielBelenky done.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by gbenhaim** [comment]
ci test please

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by ovirt-infra** [comment]
Build finished.

**Comment by gbenhaim** [comment]
ci merge please

**Comment by ovirt-infra** [comment]
Build finished.
